src/_Derivatives/subexpressions.jl

```diff
@@ -14,78 +14,130 @@ function list_subexpressions(nd::Vector{NodeData})
     return sort(collect(indices))
 end
 
-# Order the subexpressions which main_expressions depend on such that we can
-# run forward mode in this order.
-function order_subexpressions(
-    main_expressions::Vector{Vector{NodeData}},
+"""
+Return the list of dependent subexpressions. Similar to `list_subexpressions`,
+but without the unnecessary `Set{Int}` -> `Vector{Int}` -> `sort`.
```

Comment:
Document what the function does more directly. This version may contain duplicates. It's up to the caller to decide if removing duplicates is "unnecessary" or not.